pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -247,6 +247,10 @@ type ClusterSpec struct {
 
 	// Enables more verbose logging in KKP's user-cluster-controller-manager.
 	DebugLog bool `json:"debugLog,omitempty"`
+
+	// Optional: DisableCSIDriver disables the installation of CSI driver on the cluster
+	// If this is true at the data center then it can't be over-written in the cluster configuration
```

Comment:
When disableCsiDriver is true in the dc spec then we are defaulting to set it as true in the cluster spec as well which is enforced before the validation is done so even if the user tries to set it false it will get set as true before the validation is done.

Reply:
If disableCsiDriver is set to true in the dc spec then while defaulting it is set as true in the cluster spec as well so if a user tries to set it as false it will get overwritten while defaulting which happens before the validation.

Reply:
That's defaulting not validation. We need to explicitly reject it when users change this field so they know what's happening. Otherwise it will be magically default and won't be explicit for the end user.

Reply:
Added validation.
```
error: clusters.kubermatic.k8c.io "csi-test" could not be patched: admission webhook "clusters.kubermatic.io" denied the request: spec.DisableCSIDriver: Forbidden: CSI driver installation is disabled on the data center, can't be enabled on cluster
```